### DISCUSSION TITLE
how do others use HMR with code-server

### DISCUSSION THREAD

**Discussion by simonjcarr** [coder/code-server#4708] [Q&A]
I have tested code-server while listening on the public interface of my cloud VM (no reverse proxy) and I am able to serve a VueJS application in development mode and have HMR working perfectly. In the background I believe HMR is running a web socket server on another port.

When I try to do the same through a reverse proxy with code-server listening on localhost, HMR does not work. It's obvious why, the only port being forwarded is the port code server is running on. 

I could do some digging into the networking of my VueJS app and each time I start a new development server, set new reverse proxy rules, but that would be more trouble than it's worth. 

Not having a live preview of the code I am working on is a deal breaker when working with frontend JS. 

Has anyone else got a workaround for this situation?

**Comment by simonjcarr** [comment]
I have half solved the problem with the help of this tutorial

https://coder.com/docs/code-server/latest/guide#proxying-to-a-vue-app

I know have the dev server running behind the the nginx reverse proxy, but still don't have the HMR working.

For the time being I can live with a manual refresh, but will keep looking for a solution

**Reply by jsjoeio** [reply]
There may be some information worth looking at in these discussion:
- https://github.com/coder/code-server/discussions/3912#discussioncomment-1233976
- https://github.com/coder/code-server/discussions/4541#discussioncomment-1786491

Let me know if you figure it out! I've had issues with Vite & HMR but haven't had time to investigate.

**Reply by jsjoeio** [reply]
Following up here, @bpmct helped me get HMR reloading working in code-server by having me edit my `vite.config.js` settings to this:
```js
export default defineConfig({
  base: "/proxy/3000",
  server: {
    hmr: {
      clientPort: 8080,
    },
  },
});
```

And then accessing it via http://localhost:8080/proxy/3000/proxy/3000/

**Comment by Antebios** [comment]
Is there something for SolidJs?
